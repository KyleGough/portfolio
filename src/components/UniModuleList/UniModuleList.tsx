import React from 'react';

interface UniModuleListProps {
  name: string;
  children: React.ReactNode;
}

export const UniModuleList: React.FC<UniModuleListProps> = ({
  name,
  children,
}) => (
  <div className="flex flex-row align-middle my-8">
    <div className="flex items-center justify-center w-12">
      <h3 className="text-xl text-center -rotate-90 whitespace-nowrap">
        {name}
      </h3>
    </div>
    <ul className="list-none pl-4 py-4 border-l-2 rounded-xl">{children}</ul>
  </div>
);
