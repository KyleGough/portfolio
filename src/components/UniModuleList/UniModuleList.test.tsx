import { render, screen } from '@testing-library/react';
import React from 'react';

import { UniModuleList } from './UniModuleList';

describe('UniModuleList component', () => {
  it('renders', async () => {
    const { container } = render(
      <UniModuleList name="Year 3">
        <li>Machine Learning</li>
      </UniModuleList>
    );

    expect(screen.getByText('Year 3')).toBeVisible();
    expect(screen.getByText('Machine Learning')).toBeVisible();
    expect(container).toMatchSnapshot();
  });
});
