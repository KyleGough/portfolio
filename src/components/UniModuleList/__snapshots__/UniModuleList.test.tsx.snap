// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UniModuleList component renders 1`] = `
<div>
  <div
    class="flex flex-row align-middle my-8"
  >
    <div
      class="flex items-center justify-center w-12"
    >
      <h3
        class="text-xl text-center -rotate-90 whitespace-nowrap"
      >
        Year 3
      </h3>
    </div>
    <ul
      class="list-none pl-4 py-4 border-l-2 rounded-xl"
    >
      <li>
        Machine Learning
      </li>
    </ul>
  </div>
</div>
`;
